test/receiverSpec.js

```diff
@@ -288,6 +303,28 @@ describe('Receiver Module', function () {
           expect(results).to.deep.equal(['sent'])
         })
 
+        it('should ignore transfers without packet.data', function * () {
+          delete this.transfer.data.ilp_header.data
+          const request = this.receiver.createRequest({
+            amount: 1,
+            id: '22e315dc-3f99-4f89-9914-1987ceaa906d'
+          })
+          const results = yield this.client.emitAsync('incoming_prepare', this.transfer)
+          expect(results).to.deep.equal(['condition-mismatch'])
+        })
+
+        it('should fulfill conditions of transfers packet.data.expires_at but no packet.data.data', function * () {
```

Comment:
Hmm, it should fulfill requests that didn't have data in them originally but if the data is deleted from a request that should fail. Seems like this wording and the test should be changed a bit.